resources/js/Components/Form/ItemEditForm.tsx

```diff
@@ -0,0 +1,122 @@
+import {useForm} from '@inertiajs/inertia-react'
+import {Button, Grid, TextField, Typography} from '@mui/material'
+import {ErrorText} from 'Components'
+import React from 'react'
+import styled from 'styled-components'
+import {ItemEditData} from 'Types/item-data'
+import route from 'ziggy-js'
+
+type ItemEditFormPropType = {
+    onCloseDrawer: () => void
+    editData: ItemEditData
+}
+
+const StyledGrid = styled(Grid)`
+    margin-bottom: 16px;
+`
+
+const ItemEditForm = ({onCloseDrawer, editData}: ItemEditFormPropType) => {
+    const {data, setData, errors, clearErrors, put} = useForm(editData)
+
+    return (
+        <>
+            <Typography>
+                Fill out the following fields with your character&apos;s item details.
+            </Typography>
+            <Grid container spacing={0}>
+                <StyledGrid item xs={12}>
+                    <TextField
+                        margin='normal'
+                        fullWidth
+                        id='name'
+                        label='Name'
+                        name='Name'
+                        value={data.name}
+                        onChange={(e) => setData('name', e.target.value)}
+                    />
+                    {errors?.name && <ErrorText message={errors?.name} />}
+                </StyledGrid>
+                <StyledGrid item xs={12}>
+                    <TextField
+                        margin='normal'
+                        fullWidth
+                        id='description'
+                        label='Description'
+                        name='Description'
+                        value={data.description}
+                        onChange={(e) => setData('description', e.target.value)}
+                    />
+                    {errors?.description && <ErrorText message={errors?.description} />}
+                </StyledGrid>
+                <StyledGrid item xs={12}>
```

Comment:
can you use the same thing as me for rarity in `ItemForm`, rarity should be a select